app/src/main/java/org/mozilla/fenix/search/SearchController.kt

```diff
@@ -82,15 +80,19 @@ class DefaultSearchController(
 
     override fun handleTextChanged(text: String) {
         store.dispatch(SearchFragmentAction.UpdateQuery(text))
-        store.dispatch(SearchFragmentAction.ShowSearchShortcutEnginePicker(
-            text.isEmpty() && context.settings().shouldShowSearchShortcuts
-        ))
-        store.dispatch(SearchFragmentAction.ShowSearchSuggestionsHint(
-            text.isNotEmpty() &&
-                    (context as HomeActivity).browsingModeManager.mode.isPrivate &&
-                    !context.settings().shouldShowSearchSuggestionsInPrivate &&
-                    !context.settings().showSearchSuggestionsInPrivateOnboardingFinished
-        ))
+        store.dispatch(
+            SearchFragmentAction.ShowSearchShortcutEnginePicker(
+                text.isEmpty() && context.settings().shouldShowSearchShortcuts
+            )
+        )
+        store.dispatch(
+            SearchFragmentAction.ShowSearchSuggestionsHint(
+                text.isNotEmpty() &&
+                        (context as HomeActivity).browsingModeManager.mode.isPrivate &&
+                        !context.settings().shouldShowSearchSuggestionsInPrivate &&
+                        !context.settings().showSearchSuggestionsInPrivateOnboardingFinished
```

Comment:
Another out of scope, but this logic seems counter-intuitive.  I'm going to look into this.